neptune/internal/mockpoolutils.go

```diff
@@ -13,6 +13,10 @@ This module provides a handful of mock convenience functions that can be
 used to inject behaviour into NeptunePoolMock.
 */
 
+// a mock error that is judged to be not transient by neptune.isTransientError
+// which prevents retry attempts in tests
+var NonTransientErr = errors.New(" MALFORMED REQUEST ")
```

Comment:
I think there is a convention in Go where Error messages should be lower case. I can't find a definitive reference for this but I am sure I've seen this somewhere - does this get picked up by the linter?

Reply:
I think you're right, though in this case the error imitates something that comes back from the Neptune driver. There is some existing logic that identifies errors by this specific text, to prevent retries if these particular type of errors occur. 

(nothing being flagged by my linters - I wonder if it's not picked up in a test file 🤔 )

https://github.com/ONSdigital/dp-graph/blob/87a37fd21336b52a3819c93bb95170e45c5aaceb/neptune/neptune.go#L222

Reply:
I think some refactoring should be done to allow mocking the call at a higher level, so the specific error isn't required. Not sure if it's too much for this fix though.

Reply:
If this is replicating an error from an external package thats fine - only other suggestion I have is could we import and alias  that error? It saves having to reimplement it and also gets updated if they update it. If not how is is fine.